src/utils/dom.ts

```diff
@@ -27,17 +27,7 @@ export function getThemeName(el: HTMLElement): "light" | "dark" {
 }
 
 export function getElementStyleDir(el: HTMLElement): Direction {
```

Comment:
Sidebar: [not for this PR] 2 things regarding this util,

1. I think this should still go through the property. [From MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/direction#:~:text=Note%20that%20text%20direction%20is%20usually%20defined%20within%20a%20document%20(e.g.%2C%20with%20HTML%27s%20dir%20attribute)%20rather%20than%20through%20direct%20use%20of%20the%20direction%20property.):

> Note that text direction is usually defined within a document (e.g., with HTML's dir attribute) rather than through direct use of the direction property.

An additional reason would be that it may also be more performant since the browser doesn't have to compute the styles (including unrelated ones to the direction).

2. Since we're going to stick with one util, the name can be `getElementDir` as the origin of the direction is an implementation detail.

Reply:
1. I think we should use the computed style. Otherwise, if someone does set the style we wouldn't know. I see it as equivalent to the `hidden` property and `display:none`. Even though MDN advises to set the style via an attribute, it says "usually" not "should".

2. I agree that we should rename to `getElementDir`. I can do that...

Reply:
WRT 1. I get your point, but how common is setting direction via HTML vs CSS? I think this is what MDN is hinting at. The W3C also [recommends against setting direction via CSS](https://www.w3.org/International/questions/qa-html-dir#no_css):

> ### Don't use CSS for direction!
> Do _not_ use CSS to apply base direction in HTML pages.
> 
> Basically, this is because you want the directional information to be available even when the CSS is not. The directional information can affect the semantics of your content, and so should be part of the markup. (See a [longer explanation](https://www.w3.org/International/questions/qa-bidi-css-markup)).
> 
> Both the CSS and HTML specs echo this same admonition.

Reply:
I get your point too :)

But, previously, we had to pierce multiple shadow DOMs to look for a `[dir]` attribute, I'm not sure how performant that is anyway. (using .closest() crossing shadow).

It seems like we would be better off just seeing if the host has a direction style IMO.

Reply:
I guess the case of when CSS wouldn't be available would be screen readers.  Is there any other case where CSS wouldn't be available?

Reply:
> But, previously, we had to pierce multiple shadow DOMs to look for a [dir] attribute, I'm not sure how performant that is anyway. (using .closest() crossing shadow).

One key difference is that computing the style of any element will cause [layout thrashing](https://gist.github.com/paulirish/5d52fb081b3570c81e3a), traversing the DOM will not.

> I guess the case of when CSS wouldn't be available would be screen readers. 

I may have misunderstood your comment, but I don't think this is correct. CSS does affect what screen readers interpret. For example, any readable element targeted by a selector with `display: none` won't be read by screen readers. Let me know if I missed something.

Reply:
@jcfranco would you like me to put the previous function back? Its not as big of deal now that its not used as much.

Reply:
Yes, please.

Reply:
That makes sense about display:none with screen readers.  I was mostly asking that out of curiosity.  I was wondering what would be the typical use cases of concern with regards to not having CSS available.  I suppose the main benefit is that there wouldn't be any layout thrashing or FOUC (flash of unstyled content) or something similar when you don't rely on computed styles to set the UI based on direction.